coding-style.md

```diff
@@ -0,0 +1,30 @@
+# C# Coding Style
+
+We follow a similar coding style as [dotnet/runtime](https://github.com/dotnet/runtime).
+
+For non code files (xml, etc), our current best guidance is consistency. When editing files, keep new code and changes consistent with the style in the files. For new files, it should conform to the style for that component. If there is a completely new component, anything that is reasonably broadly accepted is fine.
+
+The general rule we follow is "_use Visual Studio defaults_".
+
+1. We use [Allman style](http://en.wikipedia.org/wiki/Indent_style#Allman_style) braces, where each brace begins on a new line. A single line statement block can go without braces but the block must be properly indented on its own line and must not be nested in other statement blocks that use braces (See rule 17 for more details). One exception is that a `using` statement is permitted to be nested within another `using` statement by starting on the following line at the same indentation level, even if the nested `using` contains a controlled block.
+2. We use four spaces of indentation (no tabs).
+3. We use `_camelCase` for internal and private fields and use `readonly` where possible. Prefix internal and private instance fields with `_`, static fields with `s_` and thread static fields with `t_`. When used on static fields, `readonly` should come after `static` (e.g. `static readonly` not `readonly static`).  Public fields should be used sparingly and should use PascalCasing with no prefix when used.
```

Comment:
Didn't know `s_` for static fields, will use it

Didn't find any in our source code, is it what we recommend?

Reply:
I also discovered it recently, suggested by Resharper.

Reply:
Hmm, not sure about this one since it's a form of Hungarian notation. What advantages does it bring? Static fields should really-really rare though, especially thread static ones.

BTW such a convention will violate the general naming convention of fields of avoiding snake case.

Reply:
No t_ or s_